### PR TITLE
Lazy import tensorboard

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Avoids importing tensorboard unless the logger is used.
Part of #12786
Part of #15662 

### Does your PR introduce any breaking changes? If yes, please list them.

None

cc @awaelchli @borda @akihironitta

### CODE REVIEW

File: src/pytorch_lightning/loggers/tensorboard.py [hunk 1]
```diff
@@ -188,6 +197,12 @@ def experiment(self) -> SummaryWriter:
         assert rank_zero_only.rank == 0, "tried to init log dirs in non global_rank=0"
         if self.root_dir:
             self._fs.makedirs(self.root_dir, exist_ok=True)
+
+        if _TENSORBOARD_AVAILABLE:
+            from torch.utils.tensorboard import SummaryWriter
+        else:
+            from tensorboardX import SummaryWriter  # type: ignore[no-redef]
```
Comment (Borda) [inline src/pytorch_lightning/loggers/tensorboard.py:204]: how many times is this called per epoch?

File: src/pytorch_lightning/loggers/tensorboard.py [hunk 2]
```diff
@@ -224,6 +239,12 @@ def log_hyperparams(
 
         if metrics:
             self.log_metrics(metrics, 0)
+
+            if _TENSORBOARD_AVAILABLE:
+                from torch.utils.tensorboard.summary import hparams
+            else:
+                from tensorboardX.summary import hparams  # type: ignore[no-redef]
```
Comment (Borda) [inline src/pytorch_lightning/loggers/tensorboard.py:246]: how many times is this called per epoch?

File: tests/tests_pytorch/loggers/test_all.py [hunk 3]
```diff
@@ -15,22 +15,16 @@
...
 from pytorch_lightning.demos.boring_classes import BoringModel
-from pytorch_lightning.loggers import (
-    CometLogger,
-    CSVLogger,
-    MLFlowLogger,
-    NeptuneLogger,
-    TensorBoardLogger,
-    WandbLogger,
-)
+from pytorch_lightning.loggers import CometLogger, CSVLogger, MLFlowLogger, NeptuneLogger, WandbLogger
```
Comment (Borda) [inline tests/tests_pytorch/loggers/test_all.py:25]: why not import here as all others?

File: src/pytorch_lightning/loggers/tensorboard.py (hunk 1 above)
Comment (carmocca) [reply src/pytorch_lightning/loggers/tensorboard.py:204]: Just once. Importing it again is a no-op

**Review by awaelchli — APPROVED** [review]

File: src/pytorch_lightning/loggers/tensorboard.py (hunk 1 above)
Comment (akihironitta) [reply src/pytorch_lightning/loggers/tensorboard.py:204]: @Borda Here's a quick example of importing the same thing multiple times: https://stackoverflow.com/questions/37067414/python-import-multiple-times

**Review by akihironitta — APPROVED** [review]

File: src/pytorch_lightning/loggers/tensorboard.py [hunk 4]
```diff
@@ -40,6 +38,13 @@
 log = logging.getLogger(__name__)
 
 _TENSORBOARD_AVAILABLE = RequirementCache("tensorboard")
+_TENSORBOARDX_AVAILABLE = RequirementCache("tensorboardX")
+if TYPE_CHECKING:
+    # assumes at least one will be installed when type checking
+    if _TENSORBOARD_AVAILABLE:
```
Comment (Borda) [inline src/pytorch_lightning/loggers/tensorboard.py:44]: this is opposite what @lantiga said in https://github.com/Lightning-AI/lightning/pull/15728#issuecomment-1322363710 :otter:

File: src/pytorch_lightning/loggers/tensorboard.py [hunk 5]
```diff
@@ -109,6 +114,10 @@ def __init__(
         sub_dir: Optional[_PATH] = None,
         **kwargs: Any,
     ):
+        if not _TENSORBOARD_AVAILABLE and not _TENSORBOARDX_AVAILABLE:
```
Comment (Borda) [inline src/pytorch_lightning/loggers/tensorboard.py:117]: this won't happen as TBX is mandatory depenency

File: src/pytorch_lightning/loggers/tensorboard.py (hunk 2 above)
Comment (carmocca) [reply src/pytorch_lightning/loggers/tensorboard.py:246]: This was answered in https://github.com/Lightning-AI/lightning/pull/15762#discussion_r1028655601

**Pull request merged**